crates/kitsune_p2p/kitsune_p2p/tests/peer_restart.rs

```diff
@@ -0,0 +1,147 @@
+        .configure_tx5_network(signal_url)
+        .use_bootstrap_server(bootstrap_addr);
+
+    let sender_restart = harness_restart
+        .spawn()
+        .await
+        .expect("should be able to spawn node");
+
+    let agent_restart = harness_restart.create_agent().await;
+
+    sender_restart
+        .join(space.clone(), agent_restart.clone(), None, None)
+        .await
+        .unwrap();
+
+    // Sent a message to make sure that the connection is established
+    wait_for_connected(sender_online.clone(), agent_restart.clone(), space.clone()).await;
+
+    // Wait until the connection is found
+    tokio::time::timeout(std::time::Duration::from_secs(5), {
+        let sender_online = sender_online.clone();
+        async move {
+            loop {
+                let dump = sender_online.dump_network_stats().await.unwrap();
+
+                let connection_count = connection_ids_from_dump(&dump).len();
+                if connection_count == 1 {
+                    break;
+                }
+
+                tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+            }
+        }
+    })
+    .await
+    .unwrap();
+
+    let dump = sender_online.dump_network_stats().await.unwrap();
+    let initial_connection_ids = connection_ids_from_dump(&dump);
+    assert_eq!(1, initial_connection_ids.len());
+
+    let initial_connection_id = initial_connection_ids[0].clone();
+
+    // Restart the node
+    let sender_restart = harness_restart
+        .simulated_restart(sender_restart)
+        .await
+        .expect("should be able to spawn node");
+
+    // Rejoin the space following the restart
+    sender_restart
+        .join(space.clone(), agent_restart.clone(), None, None)
+        .await
+        .unwrap();
+
+    // Wait until there is a new connection and the old one is closed
+    tokio::time::timeout(std::time::Duration::from_secs(5), {
+        let initial_connection_id = initial_connection_id.clone();
+        let sender_online = sender_online.clone();
+        async move {
+            loop {
+                let dump = sender_online.dump_network_stats().await.unwrap();
+
+                let connection_ids = connection_ids_from_dump(&dump);
+
+                let find_existing = connection_ids
+                    .iter()
+                    .find(|&id| id == &initial_connection_id);
+                let connection_count = connection_ids.len();
+                if find_existing.is_none() && connection_count == 1 {
+                    break;
+                }
+
+                tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+            }
+        }
+    })
+    .await
+    .unwrap();
+
+    let dump = sender_online.dump_network_stats().await.unwrap();
+    let connection_ids = connection_ids_from_dump(&dump);
+    assert_eq!(1, connection_ids.len());
+    assert_ne!(initial_connection_id, connection_ids[0]);
+}
+
+fn connection_ids_from_dump(dump: &Value) -> Vec<String> {
```

Comment:
Thoroughly grim, would much prefer strong types coming back from tx5 here

Reply:
:skull_and_crossbones: - [not as bad as this : )](https://github.com/holochain/holochain/blob/72596ec20e131ecd2069db9d11b08a77da553d3b/crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs#L812-L877)

Yeah... the thought here was that these stats are majorly coming from our backend go library... and if we swap out the backend, for say, the rust one, they may not have the same structure. But, maybe it'd be worth just having multiple parsers and try to munge them into the same structs so we can avoid stuff like this.